chrome/content/zotero/itemPane.js

```diff
@@ -185,6 +189,132 @@ var ZoteroItemPane = new function() {
 	}
 	
 	
+	this.translateSelectedItems = Zotero.Promise.coroutine(function* () {
+		if (_blockTranslate) {
+			_blockTranslate = false;
+			return;
+		}
+		var collectionID = _translationTarget.objectType == 'collection' ? _translationTarget.id : undefined;
+		var items = ZoteroPane_Local.itemsView.getSelectedItems();
+		if (!_translationTarget) {
+			_translationTarget = Zotero.Libraries.userLibrary;
+		}
+		for (let item of items) {
+			yield item.translate(_translationTarget.libraryID, collectionID);
+		}
+	});
+	
+	
+	this.buildTranslateSelectContextMenu = function (event) {
+		_blockTranslate = true;
+		var menu = document.getElementById('zotero-item-addTo-menu');
+		if (event.target != menu) {
+			return;
+		}
+		// Clear previous items
+		while (menu.firstChild) {
+			menu.removeChild(menu.firstChild);
+		}
+		
+		var libraries = Zotero.Libraries.getAll();
+		for (let library of libraries) {
+			if (!library.editable || library.libraryType == 'publications') {
+				continue;
+			}
+			let libraryMenu = _createTranslationMenuForTarget(library);
+			menu.appendChild(libraryMenu);
+		}
+	};
+	
+	
+	this.setTranslateButton = function(name) {
+		var label = Zotero.getString('general.addTo', name ? name : _translationTarget.name);
+		var elem = document.getElementById('zotero-feed-item-addTo-button');
+		elem.setAttribute('label', label);
+
+		var objectType = name ? 'library' : _translationTarget._objectType;
+		var imageSrc = _getCollectionImageSrc(objectType);
+		elem.setAttribute('image', imageSrc);
+	};
+	
+
+	this.setTranslationTarget = function(translationTarget) {
+		_translationTarget = translationTarget;
+		ZoteroItemPane.setTranslateButton()
+	};
+	
+		
+	this.setToggleReadLabel = function() {
+		var markRead = false;
+		var items = ZoteroPane_Local.itemsView.getSelectedItems();
+		for (let item of items) {
+			if (!item.isRead) {
+				markRead = true;
+				break;
+			}
+		}
+		var elem = document.getElementById('zotero-feed-item-toggleRead-button');
+		if (markRead) {
+			elem.setAttribute('label', Zotero.getString('pane.item.markAsRead'));
+		} else {
+			elem.setAttribute('label', Zotero.getString('pane.item.markAsUnread'));
+		}
+	};
+
+
+	/**
+	 * Create a libraryOrCollection DOM tree to place in <menupopup> element.
+	 * 
+	 * @return {Node<menuitem>/Node<menu>}
+	 */
+	function _createTranslationMenuForTarget(libraryOrCollection) {
+		var ns = "http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul";
+		function _createMenuitem(label, value, icon, command) {
+			let menuitem = document.createElementNS(ns, 'menuitem');
```

Comment:
`createElementNS` shouldn't be necessary — `document` should be a `XULDocument` already (and will be if you use `menu.ownerDocument` after moving this function).